src/cuml/clustering.jl

```diff
@@ -123,67 +124,84 @@ end
 # Multiple dispatch for initializing models
 model_init(mlj_model::KMeans) = cuml.KMeans(; mlj_to_kwargs(mlj_model)...)
 model_init(mlj_model::DBSCAN) = cuml.DBSCAN(; mlj_to_kwargs(mlj_model)...)
-model_init(mlj_model::AgglomerativeClustering) = cuml.AgglomerativeClustering(; mlj_to_kwargs(mlj_model)...)
+model_init(mlj_model::AgglomerativeClustering) =
+    cuml.AgglomerativeClustering(; mlj_to_kwargs(mlj_model)...)
 model_init(mlj_model::HDBSCAN) = cuml.HDBSCAN(; mlj_to_kwargs(mlj_model)...)
 
-const CUML_CLUSTERING = Union{KMeans, DBSCAN, AgglomerativeClustering, HDBSCAN}
+const CUML_CLUSTERING = Union{KMeans,DBSCAN,AgglomerativeClustering,HDBSCAN}
 
 # add metadata
 MMI.load_path(::Type{<:KMeans}) = "$PKG.KMeans"
 MMI.load_path(::Type{<:DBSCAN}) = "$PKG.DBSCAN"
 MMI.load_path(::Type{<:AgglomerativeClustering}) = "$PKG.AgglomerativeClustering"
 MMI.load_path(::Type{<:HDBSCAN}) = "$PKG.HDBSCAN"
 
-MMI.input_scitype(::Type{<:CUML_CLUSTERING}) = Union{AbstractMatrix, Table(Continuous)}
+MMI.input_scitype(::Type{<:CUML_CLUSTERING}) = Union{AbstractMatrix,Table(Continuous)}
+MMI.target_scitype(::Type{<:KMeans}) = AbstractVector{<:Finite}
+MMI.output_scitype(::Type{<:KMeans}) = AbstractVector{<:Finite}
```

Comment:
You don't need an `output_scitype` unless you implement a `transform`; `target_scitype` refers to `predict.